From 8fa1095bb13ea7f366ade8c4345bb3aeead259ac Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Thu, 28 Oct 2021 09:11:02 +0200
Subject: [PATCH 0298/1306] server: Clear the
 MOUSEEVENTF_(ABSOLUTE|VIRTUALDESK) flags.

For rawinput messages, as user32 is currently only expecting relative
motion.
---
 server/queue.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/server/queue.c b/server/queue.c
index 497718bea8e..c60e9ce7c5b 100644
--- a/server/queue.c
+++ b/server/queue.c
@@ -1926,7 +1926,7 @@ static int queue_mouse_message( struct desktop *desktop, user_handle_t win, cons
         msg_data = &raw_msg.data;
         msg_data->info                = input->mouse.info;
         msg_data->size                = sizeof(*msg_data);
-        msg_data->flags               = flags;
+        msg_data->flags               = flags & ~(MOUSEEVENTF_ABSOLUTE | MOUSEEVENTF_VIRTUALDESK);
         msg_data->rawinput.type       = RIM_TYPEMOUSE;
         msg_data->rawinput.mouse.x    = x - desktop->cursor.x;
         msg_data->rawinput.mouse.y    = y - desktop->cursor.y;
-- 
2.43.2

